### PR TITLE
fixed dashboard crash and patients sorting

### PR DESCRIPTION
**Pull request by shivrajp1300**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes [link to issue]

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

**Comment by sohamtriveous** [comment]
why isn't there any explanation/documentation on why the issue is happening and the approach you have taken?

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/dashboard/DashboardScreen.kt [hunk 1]
```diff
@@ -184,12 +184,6 @@ fun DashboardScreen(
                 color = GreyTextColor,
                 style = MaterialTheme.typography.h6
               )
-              Spacer(modifier = Modifier.height(4.dp))
-              Text(
-                text = "Mallapadi 1",
-                color = Color.Black,
-                style = MaterialTheme.typography.h5
-              )
```
Comment (sohamtriveous) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/dashboard/DashboardScreen.kt:192]: why is this in this release?

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/patients/RegisterViewModel.kt [hunk 2]
```diff
@@ -711,34 +711,41 @@ constructor(
...
-        if(it.extension?.get(0)?.value?.asStringValue()?.isNotEmpty() == true){
-          val date = convertToDateStringToDate(it.extension?.get(0)?.value?.asStringValue() ?: "")
-          date.isToday()
+        val extension = it?.extension?.find { it.url?.substringAfterLast("/").equals("patient-registraion-date") }
+        if(extension != null && extension.value?.asStringValue()?.isNotEmpty() == true){
+          val date = convertToDateStringToDate(extension.value?.asStringValue())
+          date?.isToday() ?: false
         }else{
-          it.meta?.lastUpdated?.isToday() == true
+          false
```
Comment (sohamtriveous) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/patients/RegisterViewModel.kt:719]: what does it mean to return false?

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/patients/RegisterViewModel.kt [hunk 3]
```diff
@@ -711,34 +711,41 @@ constructor(
       }
 
       val todayCases = patients.filter {
-
-        if(it.extension?.get(0)?.value?.asStringValue()?.isNotEmpty() == true){
-          val date = convertToDateStringToDate(it.extension?.get(0)?.value?.asStringValue() ?: "")
-          date.isToday()
+        val extension = it?.extension?.find { it.url?.substringAfterLast("/").equals("patient-registraion-date") }
+        if(extension != null && extension.value?.asStringValue()?.isNotEmpty() == true){
```
Comment (sohamtriveous) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/patients/RegisterViewModel.kt:715]: what is this if statement? can you detail what you're trying to do?

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/patients/RegisterViewModel.kt (hunk 2 above)
Comment (shivrajp1300) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/patients/RegisterViewModel.kt:719]: If patient does not have the registration date then I was using the "lastUpdated" which may have wrong date. Now I am not counting patients which doesn't have the registration date because now all patients must have a registration date.

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/patients/RegisterViewModel.kt (hunk 3 above)
Comment (shivrajp1300) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/patients/RegisterViewModel.kt:715]: Adding a check to make sure register date string is not empty otherwise date parser will throw exception.

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/dashboard/DashboardScreen.kt (hunk 1 above)
Comment (shivrajp1300) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/dashboard/DashboardScreen.kt:192]: Currently this needs to be manually updated in all sites, which was missed earlier. Removed it now.

**Comment by sohamtriveous** [comment]
this is okay but we need to write simpler methods
